frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerThread.java

```diff
@@ -304,22 +310,26 @@ public void shutdown() {
     }
 
     void setState(WorkerState newState) {
+        logger.debug("State {} -> {}", state, newState);
         listener.notifyChangeState(model.getModelName(), newState);
         long timeTaken = System.currentTimeMillis() - startTime;
-        if (state != WorkerState.WORKER_TERMINATED) {
+        if ((state != WorkerState.WORKER_SCALED_DOWN)
+                || (newState == WorkerState.WORKER_SCALED_DOWN)) {
```

Comment:
Not sure what you are doing here.
What you added case is state == WorkerState.WORKER_SCALED_DOWN and newState == WorkerState.WORKER_SCALED_DOWN
What's the point to assign the newState?

Reply:
You are correct. This is a redundant check.